engine/include/cubos/engine/ui/text/text_stretch.hpp

```diff
@@ -0,0 +1,19 @@
+/// @file
+/// @brief Component @ref cubos::engine::UITextStretch.
+/// @ingroup ui-text-plugin
+
+#pragma once
+
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/api.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Component which makes a UI element fit the text it has.
+    /// @ingroup ui-text-plugin
+    struct CUBOS_ENGINE_API UITextStretch
+    {
+        CUBOS_REFLECT;
+    };
```

Comment:
@DiogoMendonc-a is this what you envisioned?

Reply:
I added this component because sometimes we could want to have a text displayed on screen and the size of the UIElement to be the whole text, it would be annoying to need to set manually the correct size and could be prone to errors specially if the font change since different fonts would have different spacing.

This component is used in the sample, the title with "cubosengine.org" has this component because the size needs to be perfectly fit to the text to allow proper positions calculations with the anchor and pivot since these depend on the size of the element.